linux_pipeline/Jenkinsfile_ubuntu_azure_kernel_validation

```diff
@@ -0,0 +1,226 @@
+KERNEL_BOTH = "both"
+PLATFORM_HYPERV = "hyperv"
+PLATFORM_AZURE = "azure"
+PLATFORM_BOTH = "both"
+HYPERV_PARAMS_AZURE = "azure_kernel=yes; azure_kernel_edge=no"
+HYPERV_PARAMS_AZURE_EDGE = "azure_kernel=no; azure_kernel_edge=yes"
+XENIAL_VERSION = "ubuntu_16.04.3"
+BIONIC_VERSION = "Ubuntu_18.04"
+AZURE_AUTOMATION_GIT = "https://github.com/iamshital/azure-linux-automation.git"
+ARM_XENIAL = "Canonical UbuntuServer 16.04-LTS latest"
+ARM_BIONIC = "Canonical UbuntuServer 18.04-LTS latest"
+EXIT_CODE_BOTH = "10"
+EXIT_CODE_XENIAL = "11"
+EXIT_CODE_BIONIC = "12"
+
+if ("${env:distro}" == "${DISTRO_TRIGGER}") {
+    stage ("Check for kernel update") {
+        node ("ws2016"){
+            withCredentials(bindings: [string(credentialsId: 'AZURE_KERNEL_INFO_LOCATION', variable: 'AZURE_KERNEL_INFO_LOCATION')]) {
+                dir (env.BUILD_NUMBER + env.BRANCH_NAME) { 
+                    checkout scm
+                    echo 'Checking if there is a new kernel version'
+                    kernel_sts = PowerShellWrapper('''
+                        & ".\\scripts\\ubuntu_azure_kernel\\get_ubuntu_azure_kernel_version.ps1"
+                            -RemoteLocation "${env:AZURE_KERNEL_INFO_LOCATION}"
+                    ''') // LatestVersionsFile changed
+                    if ("${kernel_sts}" == "${EXIT_CODE_BOTH}") {
+                        echo "New kernel available for Bionic and Xenial. Triggering testing for both"
+                        env.distro = "${DISTRO_BOTH}"
+                        distroToTest.add('${DISTRO_BIONIC}')
+                        distroToTest.add('${DISTRO_XENIAL}')
+                    }
+                    if ("${kernel_sts}" == "${EXIT_CODE_XENIAL}") {
+                        echo "New kernel available for Xenial. Triggering testing"
+                        env.distro = "${DISTRO_XENIAL}"
+                        distroToTest.add('${DISTRO_XENIAL}')
+                    }
+                    if ("${kernel_sts}" == "${EXIT_CODE_BIONIC}") {
+                        echo "New kernel available for Bionic. Triggering testing"
+                        env.distro = "${DISTRO_BIONIC}"
+                        distroToTest.add('${DISTRO_BIONIC}')
+                    }
+                    deleteDir()
+                }
+            } 
+        }
+    }
+    if ("${kernel_sts}" == "0") {
+        echo "No new kernels found"
+        currentBuild.result = 'ABORTED'
+        return
+    }
+    if (("${kernel_sts}" != "0") || ("${kernel_sts}" != "${EXIT_CODE_BOTH}") || ("${kernel_sts}" != "${EXIT_CODE_XENIAL}") || ("${kernel_sts}" != "${EXIT_CODE_BIONIC}")) {
+        echo "get_uuntu_azure_kernel_version.ps1 failed with exit code ${kernel_sts}"
+        currentBuild.result = 'FAILED'
+        return
+    }
+} else {
+    if (("${env.kernel}" == "${KERNEL_BOTH}") || ("${env.kernel}" == "${KERNEL_AZURE}")) {
+        if ("${env:distro}" == "${DISTRO_BOTH}") {
+            distroToTest.add('${DISTRO_BIONIC}')
```

Comment:
this is groovy, never use in groovy string escaping like in bash, no need for it

Reply:
this applies for all the code in the groovy file